src/ImageManipulation.php

```diff
@@ -893,29 +930,36 @@ public function manipulate($variant, $callback)
         // Create this asset in the store if it doesn't already exist,
         // otherwise use the existing variant
         $store = Injector::inst()->get(AssetStore::class);
-        $result = null;
+        $tuple = $manipulationResult = null;
         if (!$store->exists($filename, $hash, $variant)) {
             // Circumvent generation of thumbnails if we only want to get existing ones
             if (!$this->getAllowGeneration()) {
                 return null;
             }
-            $result = call_user_func($callback, $store, $filename, $hash, $variant);
+            list($tuple, $manipulationResult) = call_user_func($callback, $store, $filename, $hash, $variant);
```

Comment:
@maxime-rainville I think this section might pose a potential BC-risk - if anyone is calling `$image->manipulate($variant, $myCustomClosure);` they’d have to update their return type. I’ll add in a check here to preserve BC